src/config/types.ts

```diff
@@ -39,7 +38,7 @@ export namespace SemverGroupConfig {
     range: SemverRange;
   }
 
-  export type Any = Union.Strict<Ignored | WithRange>;
+  export type Any = GroupConfig & Partial<Ignored & WithRange>;
```

Comment:
What was happening when you made this change? Maybe I can help. This doesn't seem right as `Ignored` and `WithRange` are mutually exclusive but this change makes them a combination of both types added together.

This comment also apples to the same change made below.

Reply:
> What was happening when you made this change?
ts-json-schema-generator doesn't understand the union type.

> This doesn't seem right as Ignored and WithRange are mutually exclusive but this change makes them a combination of both types added together.
Maybe I don't understand the union correctly. It is a combination of the types right?
The Union.Strict makes sure when type is `Ignored` all other fields of `Ignored` are also required.

What you want would be:
```suggestion
  export type Any = Ignored | WithRange;
```

Here it is not possible to define `Ignored` and `WithRange`

Reply:
Hopefully this makes sense, what `Union.Strict<Ignored | WithRange>` is describing is:

```ts
// Accept exactly one or the other of the types inside <here>
// - do not allow an Ignored with additional properties
// - do not allow a WithRange with additional properties
// - do not allow an Ignored with a WithRange on top of it
Union.Strict<
  // Accept either one or the other, but not both at the same time
  Ignored | WithRange
>
```

In contrast `Ignored & WithRange` is creating one new type, which is `Ignored` with all the properties of `WithRange` merged together on top into one object.

Maybe give it a try by going back to how it was but remove every usage of `Union.Strict` and see if everything still works. It may be fine, but I think _possibly_ there may be some cases where the problems linked [here](https://millsp.github.io/ts-toolbelt/modules/union_strict.html) may come up.